packages/website/src/theme/MDXComponents/TryInPlayground.tsx

```diff
@@ -1,20 +1,32 @@
 import Link from '@docusaurus/Link';
+import clsx from 'clsx';
 import React from 'react';
 
 import styles from './TryInPlayground.module.css';
 
 export function TryInPlayground({
   eslintrcHash,
+  children,
+  className,
+  codeHash,
 }: {
   eslintrcHash: string;
+  children?: React.ReactNode;
+  className?: string;
+  codeHash?: string;
 }): React.ReactNode {
+  const params = new URLSearchParams({ eslintrc: eslintrcHash });
+  if (codeHash) {
+    params.set('code', codeHash);
+  }
+
   return (
     <Link
-      className={styles.tryInPlaygroundLink}
-      to={`/play#eslintrc=${eslintrcHash}`}
+      className={clsx(styles.tryInPlaygroundLink, className)}
+      to={`/play#${params.toString()}`}
       target="_blank"
     >
-      Try this rule in the playground ↗
+      {children}
     </Link>
```

Comment:
[Design] The button looks great in general, but there is a bit of extra vertical spacing that I find a little awkward:

![Screenshot of the no-non-null-asserted-chain examples showing more vertical space dedicated to the new button than two existing code lines](https://github.com/typescript-eslint/typescript-eslint/assets/3335181/b29e9367-c8cc-4e3f-9771-bce8f736b11b)

I tried these styles and they reduced the vertical height a bit. What do you think?

```css
margin-top: -4rem;
position: relative;
```

![Screenshot of the same example but with much less vertical space just used for the button](https://github.com/typescript-eslint/typescript-eslint/assets/3335181/00830c5f-005f-498c-86d7-c4a86f0f071c)

Reply:
I agree for most code blocks, moving up the button a little bit looks better. But I noticed some edge cases that might need special handling:

- In some code blocks, the last line is very long. If we move the button up, then it will make the last one or two lines unreadable.
![button_over_text](https://github.com/typescript-eslint/typescript-eslint/assets/7645930/544c748f-0ea5-421c-942c-4b6e1577fc79)

- For code blocks that only contain one or two lines, the "Open in Playground" may overlap the copy button on the top right corner.
![buttons_overlap](https://github.com/typescript-eslint/typescript-eslint/assets/7645930/251bef37-00ef-4a8c-8499-f2d175eb375f)

My two cents on how to solve these issues:

- If the last one or two lines in a code block is too long, we can reformat the code a little bit, especially if it is a long comment line which can be broken into multiple lines without affecting the semantics of the original code.
- In `website/src/theme/CodeBlock/Content/String.tsx`, we can check if the code snippet only has one or two lines or the last couple of lines are too long. If not, we can add a special class name to the button to move it a little up.
- On mobile screens, I think it might be better to never move the button up due to the limited horizontal space.

What do you think?

Reply:
Blurgh yeah. I do dislike these edge cases.

> we can reformat the code a little bit

Yeah. Long comment lines feel difficult to read for docs to me anyway, and probably something to avoid. Maybe if we just move them all to a comment above the code, that'll get us most of the way?

```diff
+ // etc etc etc etc etc etc etc
+ let something;
- let something; // etc etc etc etc etc etc etc
```

Reply:
Updated in 0f7d581e785f6530b8fe7c0f6e2d51557ffb9821 and e2ed8608dad951f793cf7f8ba51a04137536ccfd.

I reformatted one .md file to shorten the last lines of a few code blocks and tweaked the style of the playground button. Now the button will not overlap the copy button and its position will be dynamically adjusted based on the screen size.

Btw, I am not sure why one of the CI check is still in progress for ~2 hours.